tinygrad/runtime/ops_cuda.py

```diff
@@ -52,12 +52,22 @@ def __init__(self, size, dtype): super().__init__(size, dtype, cuda.mem_alloc(si
     def _copyin(self, x:np.ndarray, stream:Optional[cuda.Stream]=None): cuda.memcpy_htod_async(self._buf, x.ravel(), stream) # type: ignore
     def _copyout(self, x:np.ndarray): cuda.memcpy_dtoh(x, self._buf) # type: ignore
 
+CUDA_PROGRAM_HEADER = 'struct __attribute__((device_builtin)) uint3{unsigned int x, y, z;};struct __attribute__((device_builtin)) dim3{unsigned int x, y, z;__attribute__((host)) __attribute__((device)) constexpr dim3(unsigned int vx=1,unsigned int vy=1, unsigned int vz=1):x(vx),y(vy),z(vz){} __attribute__((host)) __attribute__((device)) constexpr dim3(uint3 v):x(v.x),y(v.y),z(v.z){} __attribute__((host)) __attribute__((device)) constexpr operator uint3(void) const {return uint3{x,y,z};}};uint3 __attribute__((device_builtin)) extern const threadIdx;uint3 __attribute__((device_builtin)) extern const blockIdx;dim3 __attribute__((device_builtin)) extern const blockDim;dim3 __attribute__((device_builtin)) extern const gridDim;int __attribute__((device_builtin)) extern const warpSize;struct __attribute__((device_builtin)) float1{float x;};struct __attribute__((device_builtin)) __attribute__((aligned(8))) float2 { float x; float y; };struct __attribute__((device_builtin)) float3{float x, y, z;};struct __attribute__((device_builtin)) __attribute__((aligned(16))) float4{float x, y, z, w;};static __inline__ __attribute__((host)) __attribute__((device)) float1 make_float1(float x){float1 t; t.x = x; return t;}static __inline__ __attribute__((host)) __attribute__((device)) float2 make_float2(float x, float y){float2 t; t.x = x; t.y = y; return t;}static __inline__ __attribute__((host)) __attribute__((device)) float3 make_float3(float x, float y, float z){float3 t; t.x = x; t.y = y; t.z = z; return t;}static __inline__ __attribute__((host)) __attribute__((device)) float4 make_float4(float x, float y, float z, float w){float4 t; t.x = x; t.y = y; t.z = z; t.w = w; return t;}'
```

Comment:
Hmm, do we need all these things? What if we just don't use them?

Reply:
We need all the variables and types that are used in the kernel, otherwise it won't compile. I did include some things we are currently not using for future compatibility, but we can remove it and shorten the header to: `CUDA_PROGRAM_HEADER = 'struct __attribute__((device_builtin)) uint3{unsigned int x, y, z;};struct __attribute__((device_builtin)) dim3{unsigned int x, y, z;__attribute__((host)) __attribute__((device)) constexpr operator uint3(void) const {return uint3{x,y,z};}};uint3 __attribute__((device_builtin)) extern const threadIdx;uint3 __attribute__((device_builtin)) extern const blockIdx;struct __attribute__((device_builtin)) __attribute__((aligned(16))) float4{float x, y, z, w;}; static __inline__ __attribute__((device)) float4 make_float4(float x, float y, float z, float w){float4 t; t.x = x; t.y = y; t.z = z; t.w = w; return t;}'`

Reply:
I do think it makes more sense to keep the header as small as possible, we can always add more to it later.

Reply:
The snipped above is insufficent. The last commit I made includes the minimum number of definitions to pass CI.